changes to 02-06

nuosv opened the pull request:
closes #7

nuosv committed:
first commit to this branch

nuosv committed:
I know right..

nuosv committed:
Merge branch 'fixed-notebooks-up-to-02-04'

nuosv committed:
Fixed notebooks for new module versions

KishManani commented:
Hi @nuosv! Thanks for this contribution! Your effort is really appreciated. We're currently updating the notebooks ourselves. I'll review and test these also to see if we can merge them with minimal changes. Thanks again!

nuosv committed:
updating notebook  02-06

KishManani commented:
Hi @nuosv! Thanks for the contributions, I'll find a way to get them merged. I've noticed the same files are altered in this PR as well as this PR you raised: https://github.com/trainindata/forecasting-with-machine-learning/pull/7.

Would it be possible in each PR to write a short description of the goals of the PR and what the difference between these two PRs are? Thank you.

KishManani commented:
**Review for commits made to section 1:**

@nuosv - did you intend to commit changes to section 1? Here I noticed you primarily changed the reference to hourly resampling from `H` to `h` as well as adjust some of the creation of lag features by using freq = "7d" rather than "6d". Were they bugs? It all seems to be working as expected. @solegalli, when you get some time, would you be able to comment whether there is a typo in notebook `01-feature-engineering.ipynb` and the other notebooks in section when setting `freq="6d"` or should it be `freq="7d"`? Thank you!   

If there are bugs we can fix them, but we need to re-run the notebook once from top to bottom so the cell outputs are displayed correctly before we can merge them.

KishManani commented:
**Review for 01-recursive-forecasting.ipynb**

- Do we need to specify `h` rather than `H` when resampling?
- Can we double check why we originally used 6 * 24 rather than 7 * 24 when creating a weekly lag? @solegalli, when you get some time, would you be able to comment whether there is a typo in notebook or whether the existing implementation is correct as intended? Thank you!
- The other changes look good to me!

nuosv commented:
Hi KishManani, regarding the h and H, the reason I changed it to an h was because of this warning:
"""
C:\Users\nuosv\AppData\Local\Temp\ipykernel_26364\3625115944.py:26: FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead.
  .resample("H")
"""
I also changed 144 to 168, because a week is generally 7 days and 7 days is 168 hours. 

Best regards,
Nuno Svensson.

KishManani commented:
> Hi KishManani, regarding the h and H, the reason I changed it to an h was because of this warning: """ C:\Users\nuosv\AppData\Local\Temp\ipykernel_26364\3625115944.py:26: FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead. .resample("H") """ I also changed 144 to 168, because a week is generally 7 days and 7 days is 168 hours.
> 
> Best regards, Nuno Svensson.

Thanks Nuno regarding the deprecation warning! Yes, I understand the change regarding what a week is :D -- and it looks correct, but I wanted to check with @solegalli whether the intention was different here. Thanks again!

KishManani commented:
**Review for 02-recursive-forecasting.ipynb**
- Can confirm notebook runs successfully
- Just need to confirm lag of 6 vs 7 days from @solegalli, otherwise looks good

**Review for 03-recursive-forecasting-with-window-features.ipynb**
- Can confirm notebook runs successfully, looks good to merge
- Minor suggestion, feel free to ignore, rename `rolling` to `rolling_features` so that it is more explicit.

**Review for 04-direct-forecasting-skforecast.ipynb**
- Can confirm notebook runs successfully and changes are good, looks good to merge

**Review for 05-direct-forecasting-sklearn.ipynb**
- Can confirm notebook runs successfully and changes are good, looks good to merge

**Review for 06-direct-forecasting-with-future-known-features.ipynb**
- Can confirm notebook runs successfully and changes are good, looks good to merge

Thanks @nuosv! That's really helpful. We're close to merging. We just need:
- To re-run the section 1 notebooks once from top to bottom and then push them here
- Address the minor suggestion above (optional)
- @solegalli's confirmation that we can move the lag features from 6 days to 7 days

solegalli committed:
remove uneccesary files

solegalli committed:
rerun notebook 1

solegalli committed:
rerun notebook 2

solegalli committed:
revised section 2 on multistep forecasting

solegalli commented:
> **Review for commits made to section 1:**
> 
> @nuosv - did you intend to commit changes to section 1? Here I noticed you primarily changed the reference to hourly resampling from `H` to `h` as well as adjust some of the creation of lag features by using freq = "7d" rather than "6d". Were they bugs? It all seems to be working as expected. @solegalli, when you get some time, would you be able to comment whether there is a typo in notebook `01-feature-engineering.ipynb` and the other notebooks in section when setting `freq="6d"` or should it be `freq="7d"`? Thank you!
> 
> If there are bugs we can fix them, but we need to re-run the notebook once from top to bottom so the cell outputs are displayed correctly before we can merge them.

Yes, I intended to do 7 days (1 week) windows and instead I typed 6. My mistake.

solegalli commented:
Hi @nuosv 

Thank you so much for updating the notebooks. You did a great job. I pushed some changes to your branch, where I rerun the notebooks so that the first cell is 1 or 2, and also added a few notes for those students who enrolled before the skforecast update, so that they understand what changed in the new version.

@KishManani : changed rolling for rolling_features as suggested.

If you (@nuosv) merge this PR: https://github.com/nuosv/forecasting-with-machine-learning/pull/2  in your fork, it will be automatically updated here, and then we can merge and close the issue.

Thank you so much for your work, and I truly apologise for my delay in replying.

nuosv committed:
Merge pull request #2 from trainindata/nunosv

rerun notebooks and add a few notes

solegalli committed:
Delete 02-multistep-forecasting/training.ipynb

Not sure where this file came from.

solegalli commented:
Thank you @nuosv ! I really appreciate your help with updating the code!
